src/extractor/graph_compressor.cpp

```diff
@@ -217,6 +240,14 @@ void GraphCompressor::Compress(const std::unordered_set<NodeID> &barrier_nodes,
                 graph.GetEdgeData(forward_e1).duration += forward_duration2;
                 graph.GetEdgeData(reverse_e1).duration += reverse_duration2;
 
+                if (node_weight_penalty && node_duration_penalty)
+                {
+                    graph.GetEdgeData(forward_e1).weight += *node_weight_penalty;
```

Comment:
Additional weight penalties will be lost after update of any segment of the edge-based graph node, so it is possible to get inconsistent weights values.